src/QMCDrivers/QMCDriverFactory.cpp

```diff
@@ -91,17 +92,37 @@ QMCDriverFactory::DriverAssemblyState QMCDriverFactory::readSection(xmlNodePtr c
   infoSummary.flush();
   infoLog.flush();
 
+  // Really by position if you don't write <qmc name="xxx",,,>
+  // you can write <vmc ...> so hacky
   if (curName != "qmc")
     qmc_mode = curName;
 
   const int nchars = qmc_mode.size();
+
+  // Begin to separate batch input reading from the legacy input parsing
+  if (project_data_.get_driver_epoch() == "batched")
```

Comment:
Could you add a bool argument (force_batch) of readSection for and expand unit tests of readSection.
Also check `getEngineName()` after the driver being created in unit tests.

Reply:
Since you cannot create a QMCDriverFactory without ProjectData and the global driver epoch is set there, I don't think its a good idea to also allow it to be forced directly.  There is one source of truth for this parameter and it is from the ProjectData the driver factory is constructed with.

The _batch are supported for the convenience of users and tests currently using them but are not valid input when the driver epoch is set.  Based on the driver epoch input, the qmcsystem and hamiltonian node will be parsed differently so we don't want to encourage further use of the _batch or introduce  _legacy as a tag for drivers.

Reply:
I forgot that ProjectData is required by QMCDriverFactory. Then there is no need to add a bool argument. I have not read your latest commit but I guess you have both epoch cases covered in the unit test now. What did you mean by "the qmcsystem and hamiltonian node will be parsed differently" did you mean "hamiltonian of qmcsystem" ?

Reply:
We are also going to allow an `<estimators>` node for global estimator definitions.  This will only be valid if in the the batched epoch.